third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -564,6 +564,22 @@ impl<'env> Generator<'env> {
                     self.internal_error(id, "inconsistent type in select expression")
                 }
             },
+            Operation::Exists(None)
+            | Operation::BorrowGlobal(_)
+            | Operation::MoveFrom
+            | Operation::MoveTo
+                if self.env().get_node_instantiation(id)[0]
+                    .get_struct(self.env())
+                    .is_none() =>
+            {
+                self.error(
+                    id,
+                    format!(
+                        "expected `Type::Struct`, found: `{:?}`",
```

Comment:
Compiler V1 gives a much more verbose error message here, such as:
```
Error: error[E04009]: expected specific type
  ┌─ TEMPFILE:4:17
  │
4 │         assert!(exists<T>(addr), 0);
  │                 ^^^^^^^^^^^^^^^
  │                 │      │
  │                 │      Expected a struct type. Global storage operations are restricted to struct types declared in the current module. Found the type parameter: 'T'
  │                 Invalid call to exists.
```
Can we add some explanation here, too?  Note you can use 
```
            self.env.diag_with_labels(Severity::Error, &loc, &msg, error_details);
```
where `loc = self.env.get_node_loc(id)` and `error_details: Vec<(Loc, String)>` to add some further info.  Or maybe `diag_with_notes` if we just want to add some supporting info without particular location for it.

Reply:
done

Reply:
Compiler v1 error messages are way to verbose (this is not only my idea but user feedback). We do not add explanations anywhere at this point. If we have the have them, I would like to have them separated from actual error and be optionally activated. We need to go back here at some point and remove the 'tutorial', it is inconsistent with the current messages.